skiko/build.gradle.kts

```diff
@@ -354,6 +368,75 @@ project.tasks.register<Exec>("wasmCompile") {
     outputs.files(outJs, outWasm)
 }
 
+fun List<String>.findAllFiles(suffix: String): List<String> = this
+    .map { File(it) }
+    .flatMap { it.walk().toList() }
+    .map { it.absolutePath }
+    .filter { it.endsWith(suffix) }
+
+
+// Very hacky way to compile native bridges and add the
+// resulting object files into the final native klib.
+project.tasks.register<Exec>("nativeBridgesCompile") {
+    dependsOn(skiaDir)
+    val inputDirs = listOf(
+        "$projectDir/src/macosX64Main/cpp/generated",
+        "$projectDir/src/macosX64Main/cpp/common",
+        "$projectDir/src/commonMain/cpp/common"
+    )
+    val outDir = "$buildDir/nativeBridges/obj/$target"
+    val srcs = inputDirs
+        .findAllFiles(".cc")
+        .toTypedArray()
+    val outs = srcs
+        .map { it.substringAfterLast("/") }
+        .map { File(it).nameWithoutExtension }
+        .map { "$outDir/$it.o" }
+        .toTypedArray()
+
+    workingDir = File(outDir)
+    commandLine = listOf(
+        "clang++",
+        *targetArch.clangFlags,
+        "-mmacosx-version-min=10.13",
+        "-std=c++17",
+        "-c",
+        "-DPROVIDE_JNI_TYPES",
+        "-DSK_SHAPER_CORETEXT_AVAILABLE",
+        "-DSK_BUILD_FOR_MAC",
+        "-DSK_METAL",
+        "-I$projectDir/src/macosX64Main/cpp/headers",
+        "-I$projectDir/src/commonMain/cpp/headers",
+        *skiaPreprocessorFlags,
+        *srcs
+    )
+    file(outDir).mkdirs()
+    inputs.files(srcs)
+    outputs.files(outs)
+}
+
+project.tasks.register<Exec>("nativeBridgesLink") {
+    dependsOn(project.tasks.getByName("nativeBridgesCompile"))
+    inputs.files(project.tasks.getByName("nativeBridgesCompile").outputs)
+
+    val outDir = "$buildDir/nativeBridges/static/$target"
+    val srcs = inputs.files.files
+        .map { it.absolutePath }
+        .toTypedArray()
+    println("SRCS: $srcs")
+    val staticLib = "$outDir/skiko-native-bridges-$target.a"
+    workingDir = File(outDir)
+    commandLine = listOf(
+        "libtool",
```

Comment:
Why not just ar directly?

Reply:
However I tried, all binaries produced with `ar` are ignored by linker. The binaries produced with` libtool` just work.

Reply:
Ah, maybe you need to run ranlib, but anyway - it's simpler this way then.